Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -252,15 +252,107 @@ function Wait-ServiceFabricApplicationUpgradeAction
 
     $global:operationId = $SF_Operations.WaitApplicationUpgradeStatus
 
-    Write-Host (Get-VstsLocString -Key SFSDK_WaitingForUpgrade)
-    $upgradeStatusFetcher = { Get-ServiceFabricApplicationUpgrade -ApplicationName $ApplicationName }
-    $upgradeRetryEvaluator = { param($upgradeStatus) return ($upgradeStatus.UpgradeState -ne "RollingBackCompleted" -and $upgradeStatus.UpgradeState -ne "RollingForwardCompleted") }
-    return Invoke-ActionWithRetries -Action $upgradeStatusFetcher `
-        -ResultRetryEvaluator $upgradeRetryEvaluator `
+    $upgradeStatusFetcher = {
+        param(
+            [object]$LastUpgradeStatus
+        )
+        $upgradeStatus = Get-ServiceFabricApplicationUpgrade -ApplicationName $ApplicationName
+        Write-Host (Get-VstsLocString -Key SFSDK_CurrentUpgradeState) $upgradeStatus.UpgradeState
+
+        $currentDomainWiseUpgradeStatus = Get-DomainUpgradeStatus -UpgradeDomainsStatus $upgradeStatus.UpgradeDomainsStatus
+        $lastDomainWiseUpgradeStatus = ""
+        if ($LastUpgradeStatus -ne $null)
```

Comment:
minor: you can move this if condition to inside Get-DomainUpgradeStatus. the caller will be cleaner then.